packages/flutter_tools/lib/src/temp_directory_validator.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'base/file_system.dart';
+import 'doctor_validator.dart';
+
+class TempDirectoryValidator extends DoctorValidator {
+  TempDirectoryValidator({
+    required FileSystem fileSystem
+  }) : _fileSystem = fileSystem, super('Temp Directory');
+
+  final FileSystem _fileSystem;
+
+  @override
+  String get slowWarning => 'Temporary Directory Validator check is taking a long time...';
+
+  @override
+  Future<ValidationResult> validate() async {
+    final List<ValidationMessage> messages = <ValidationMessage>[];
+    try {
+      final Directory tempDir = tempDirectory;
```

Comment:
Would calling this getter throw an exception if the directory didn't exist? I thought we were throwing an exception later, when we try to write to it.

Reply:
This getter is writing `flutter_tools` inside the temp directory thats why is failing from this point https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/base/file_system.dart#L221

Reply:
ahh, i see

Reply:
Rather than calling the getter and catching the exception, can you instead explicitly check if `fs.systemTempDirectory.existsSync()`? Using exceptions for control flow is generally an anti-pattern; I also think the code would be more clear if we explicitly check for what we are interested in, rather than a side effect from the getter implementation that might change across a refactor.